tests/ops/functions/test_iterative_qpe.py

```diff
@@ -0,0 +1,242 @@
+        sample_list = sample_list.T
+        output = qml.probs().process_samples(np.array(sample_list), wire_order=[0, 1, 2])
+
+        @qml.qnode(dev)
+        def circuit_qpe():
+            # Initial state
+            qml.PauliX(wires=[0])
+
+            # Iterative QPE
+            qml.QuantumPhaseEstimation(qml.RZ(phi, wires=[0]), estimation_wires=[1, 2, 3])
+
+            return qml.probs(wires=[1, 2, 3])
+
+        assert np.allclose(np.round(output, 2), np.round(circuit_qpe(), 2))
+
+    @pytest.mark.jax
+    def test_check_gradients_jax(self):
+        """Test to check that the gradients are correct comparing with the expanded circuit using JAX"""
+
+        import jax
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit(theta):
+            meas = qml.iterative_qpe(qml.RZ(theta, wires=[0]), [1], iters=2)
+            return qml.expval(meas[0])
+
+        @qml.qnode(dev)
+        def manual_circuit(phi):
+            qml.Hadamard(wires=[1])
+            qml.ctrl(qml.RZ(phi, wires=[0]) ** 2, control=[1])
+            qml.Hadamard(wires=[1])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[2, 1])
+            qml.Hadamard(wires=[1])
+            qml.ctrl(qml.RZ(phi, wires=[0]), control=[1])
+            qml.ctrl(qml.PhaseShift(-np.pi / 2, wires=[1]), control=[2])
+            qml.Hadamard(wires=[1])
+            qml.CNOT(wires=[1, 3])
+            qml.CNOT(wires=[3, 1])
+
+            return qml.expval(qml.Hermitian([[0, 0], [0, 1]], wires=3))
+
+        phi = jax.numpy.array(1.0)
+        assert jax.numpy.isclose(jax.grad(circuit)(phi), jax.grad(manual_circuit)(phi))
+
+    @pytest.mark.torch
+    def test_check_gradients_torch(self):
+        """Test to check that the gradients are correct comparing with the expanded circuit using PyTorch"""
+
+        import torch
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit(theta):
+            meas = qml.iterative_qpe(qml.RZ(theta, wires=[0]), [1], iters=2)
+            return qml.expval(meas[0])
+
+        @qml.qnode(dev)
+        def manual_circuit(phi):
+            qml.Hadamard(wires=[1])
+            qml.ctrl(qml.RZ(phi, wires=[0]) ** 2, control=[1])
+            qml.Hadamard(wires=[1])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[2, 1])
+            qml.Hadamard(wires=[1])
+            qml.ctrl(qml.RZ(phi, wires=[0]), control=[1])
+            qml.ctrl(qml.PhaseShift(-np.pi / 2, wires=[1]), control=[2])
+            qml.Hadamard(wires=[1])
+            qml.CNOT(wires=[1, 3])
+            qml.CNOT(wires=[3, 1])
+
+            return qml.expval(qml.Hermitian([[0, 0], [0, 1]], wires=3))
+
+        phi = torch.tensor(1.0, requires_grad=True)
+        assert torch.isclose(torch.func.grad(circuit)(phi), torch.func.grad(manual_circuit)(phi))
+
+    @pytest.mark.xfail(reason="See https://github.com/PennyLaneAI/pennylane/issues/4825")
```

Comment:
Since Matt's PR is merged now, does the test still need to xfail?

Reply:
The error was discovered to be another. We had the possibility to make Torch or TF work and by preference Torch was chosen. @timmysilv  is aware of this